storage/api.go

```diff
@@ -147,3 +135,85 @@ type GovernanceData struct {
 	ProposalFinalizations []*governance.Proposal
 	Votes                 []*governance.VoteEvent
 }
+
+// RuntimeSourceStorage defines an interface for retrieving raw block data
+// from the runtime layer.
+type RuntimeSourceStorage interface {
+	// BlockData gets block data in the specified round. This includes all
+	// block header information, as well as transactions and events included
+	// within that block.
+	BlockData(ctx context.Context, round uint64) (*RuntimeBlockData, error)
+
+	// CoreData gets data in the specified round emitted by the `core` module.
+	CoreData(ctx context.Context, round uint64) (*CoreData, error)
+
+	// AccountsData gets data in the specified round emitted by the `accounts` module.
+	AccountsData(ctx context.Context, round uint64) (*AccountsData, error)
+
+	// ConsensusAccountsData gets data in the specified round emitted by the `consensusaccounts` module.
+	ConsensusAccountsData(ctx context.Context, round uint64) (*ConsensusAccountsData, error)
+
+	// Name returns the name of the source storage.
+	Name() string
+}
+
+// RuntimeBlockData represents data for a runtime block during a given round.
+type RuntimeBlockData struct {
+	Round uint64
+
+	BlockHeader             *block.Block
```

Comment:
Hmm, I recognize that is more consistent with the consensus analyzer, but I guess technically `BlockHeader` is a `Block`.